### PR TITLE
rip out unused source plugin id checking

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
In doing some rework in support of the Incidents Service client API, I found what appears to be legacy code. Removing it simplifies things for the cleanup, but I'd like to do it out of band for clarity.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-32495

### CODE REVIEW

**Review by crspeller — APPROVED** [review]
That code was for handling plugin to plugin communication. Copy paste from other plugins that where using it. If we do use that in the future we can just add the ID to the context in `server/plugin.go` so removing it is fine.

File: server/plugin.go [hunk 1]
```diff
@@ -43,9 +43,9 @@ type Plugin struct {
 	bot                 *bot.Bot
 }
 
-// ServeHTTP demonstrates a plugin that handles HTTP requests by greeting the world.
+// ServeHTTP routes incoming HTTP requests to the plugin's REST API.
```
Comment (agarciamontoro) [inline server/plugin.go:46]: Good catch :joy:

**Review by agarciamontoro — APPROVED** [review]
Yay!

**Comment by lieut-data** [comment]
@prapti, I can't (yet) assign you to this pull request (due to the accidental org reset), but heads up that this one should just requires a quick smoketest. (Something I think the e2e tests actually already do well in this case.)

**Comment by lieut-data** [comment]
This PR will require merging with upstream before merging.

**Comment by lieut-data** [comment]
@prapti, for testing purposes, there really isn't anything "special" here that shouldn't already be covered by the unit tests. Your call on the coverage :)

**Review by prapti — APPROVED** [review]
Ran a quick smoke test and everything is passing as expected.

**Pull request merged**